lib/fluent/plugin/in_kafka_group.rb

```diff
@@ -244,49 +248,102 @@ def reconnect_consumer
     end
   end
 
+  def process_batch_with_record_tag(batch)
+    es = {} 
+    batch.messages.each { |msg|
+      begin
+        record = @parser_proc.call(msg)
+        tag = record[@record_tag_key]
```

Comment:
Ditto.

Reply:
@cosmo0920 Not sure if we can use it. I basically copied the syntax from the existing `record_time_key` option. If we change this here, shouldn't then the syntax for `record_time_key` (lines 266-270) als be changed?

Reply:
@cosmo0920 one more thing that just comes to my mind: when we are reading the records from Kafka at this stage, they are not yet Fluentd messages, so would the record_accessor helper even work?

Reply:
If we use Fluentd v1 API and handle nested records, it works. But this  plugin uses old Fluentd API. My suggestion does not work....